src/nvim/CMakeLists.txt

```diff
@@ -265,7 +283,7 @@ foreach(sfile ${NVIM_SOURCES}
   get_preproc_output(PREPROC_OUTPUT ${gf_i})
 
   set(depends "${HEADER_GENERATOR}" "${sfile}")
-  if("${f}" STREQUAL "version.c")
+  if(IS_INSIDE_GIT_TREE AND "${f}" STREQUAL "version.c")
```

Comment:
I wanted to avoid re-generating `versiondef_git.h` on every build for when (git is not available, or not a repo clone), and the easiest way was to basically drop it from being a dependency in those cases.

Reply:
tarball or non-git repo is not the common case so it doesn't matter. What matters is fixing https://github.com/neovim/neovim/pull/19289#discussion_r925717296 and keeping the code simple and clear.